### ISSUE TITLE
relational-record-examples not compiling

### ISSUE THREAD

**Issue by cdepillabout** [khibino/haskell-relational-record#6]
I am currently following along the [tutorial](http://khibino.github.io/haskell-relational-record/tutorial.html), but I am not able to compile the code from relational-record-examples.  I have tried both with relational-record-examples-0.2.0.0 from Hackage, and with relational-record-examples from the master branch of this repository.  They fail with the same error.

Here is the error:

```
me@machine ~/git/relational-record-examples-0.2.0.0 $ cabal install -j
Resolving dependencies...
Notice: installing into a sandbox located at
/home/me/git/relational-record-examples-0.2.0.0/.cabal-sandbox
Configuring dlist-0.7.1...
Configuring names-th-0.1.0.0...
Building names-th-0.1.0.0...
Building dlist-0.7.1...
Installed names-th-0.1.0.0
Configuring sql-words-0.1.3.0...
Configuring text-1.2.0.4...
Installed dlist-0.7.1
Building sql-words-0.1.3.0...
Building text-1.2.0.4...
Configuring time-locale-compat-0.1.0.1...
Installed sql-words-0.1.3.0
Building time-locale-compat-0.1.0.1...
Configuring transformers-0.4.3.0...
Installed time-locale-compat-0.1.0.1
Building transformers-0.4.3.0...
Configuring utf8-string-1...
Installed transformers-0.4.3.0
Building utf8-string-1...
Installed utf8-string-1
Configuring mtl-2.2.1...
Building mtl-2.2.1...
Configuring persistable-record-0.1.0.0...
Installed mtl-2.2.1
Building persistable-record-0.1.0.0...
Installed persistable-record-0.1.0.0
Installed text-1.2.0.4
Configuring convertible-1.1.0.0...
Configuring relational-query-0.4.0.1...
Building convertible-1.1.0.0...
Building relational-query-0.4.0.1...
Installed convertible-1.1.0.0
Configuring HDBC-2.4.0.0...
Building HDBC-2.4.0.0...
Installed relational-query-0.4.0.1
Configuring relational-schemas-0.1.0.1...
Building relational-schemas-0.1.0.1...
Installed HDBC-2.4.0.0
Configuring HDBC-session-0.1.0.0...
Downloading HDBC-sqlite3-2.3.3.0...
Building HDBC-session-0.1.0.0...
Configuring HDBC-sqlite3-2.3.3.0...
Installed HDBC-session-0.1.0.0
Building HDBC-sqlite3-2.3.3.0...
Installed HDBC-sqlite3-2.3.3.0
Installed relational-schemas-0.1.0.1
Configuring relational-query-HDBC-0.1.0.0...
Building relational-query-HDBC-0.1.0.0...
Installed relational-query-HDBC-0.1.0.0
Configuring relational-record-examples-0.2.0.0...
Building relational-record-examples-0.2.0.0...
Failed to install relational-record-examples-0.2.0.0
Build log ( /home/d_gosnell/git/relational-record-examples-0.2.0.0/.cabal-sandbox/logs/relational-record-examples-0.2.0.0.log ):
Configuring relational-record-examples-0.2.0.0...
Building relational-record-examples-0.2.0.0...
Preprocessing executable 'examples' for relational-record-examples-0.2.0.0...
[1 of 9] Compiling DataSource       ( src/DataSource.hs, dist/dist-sandbox-3657a918/build/examples/examples-tmp/DataSource.o )
[2 of 9] Compiling Employee         ( src/Employee.hs, dist/dist-sandbox-3657a918/build/examples/examples-tmp/Employee.o )
Loading package ghc-prim ... linking ... done.
Loading package integer-gmp ... linking ... done.
Loading package base ... linking ... done.
Loading package array-0.4.0.1 ... linking ... done.
Loading package deepseq-1.3.0.1 ... linking ... done.
Loading package bytestring-0.10.0.2 ... linking ... done.
Loading package containers-0.5.0.0 ... linking ... done.
Loading package transformers-0.4.3.0 ... linking ... done.
Loading package mtl-2.2.1 ... linking ... done.
Loading package old-locale-1.0.0.5 ... linking ... done.
Loading package old-time-1.1.0.1 ... linking ... done.
Loading package text-1.2.0.4 ... linking ... done.
Loading package time-1.4.0.1 ... linking ... done.
Loading package convertible-1.1.0.0 ... linking ... done.
Loading package utf8-string-1 ... linking ... done.
Loading package HDBC-2.4.0.0 ... linking ... done.
Loading package HDBC-session-0.1.0.0 ... linking ... done.
Loading package pretty-1.1.1.0 ... linking ... done.
Loading package template-haskell ... linking ... done.
Loading package names-th-0.1.0.0 ... linking ... done.
Loading package dlist-0.7.1 ... linking ... done.
Loading package persistable-record-0.1.0.0 ... linking ... done.
Loading package sql-words-0.1.3.0 ... linking ... done.
Loading package time-locale-compat-0.1.0.1 ... linking ... done.
Loading package relational-query-0.4.0.1 ... linking ... done. 
Loading package relational-schemas-0.1.0.1 ... linking ... done.
Loading package relational-query-HDBC-0.1.0.0 ... linking ... done.
Loading package HDBC-sqlite3-2.3.3.0 ... linking ... done.
[3 of 9] Compiling Transaction      ( src/Transaction.hs, dist/dist-sandbox-3657a918/build/examples/examples-tmp/Transaction.o )
[4 of 9] Compiling Product          ( src/Product.hs, dist/dist-sandbox-3657a918/build/examples/examples-tmp/Product.o )
[5 of 9] Compiling Department       ( src/Department.hs, dist/dist-sandbox-3657a918/build/examples/examples-tmp/Department.o )
[6 of 9] Compiling Customer         ( src/Customer.hs, dist/dist-sandbox-3657a918/build/examples/examples-tmp/Customer.o )
[7 of 9] Compiling Branch           ( src/Branch.hs, dist/dist-sandbox-3657a918/build/examples/examples-tmp/Branch.o )
[8 of 9] Compiling Account          ( src/Account.hs, dist/dist-sandbox-3657a918/build/examples/examples-tmp/Account.o )
[9 of 9] Compiling Main             ( src/examples.hs, dist/dist-sandbox-3657a918/build/examples/examples-tmp/Main.o )

src/examples.hs:370:29:
    Couldn't match type `Maybe Int64' with `Int64'
    Expected type: Pi Account Int64
      Actual type: Pi Account (Maybe Int64)
    In the second argument of `(!)', namely Account.accountId' 
    In the second argument of `(|$|)', namely a ! Account.accountId'
    In the first argument of `(|*|)', namely
      Account1 |$| a ! Account.accountId'

src/examples.hs:371:29:
    Couldn't match type `Maybe String' with `[Char]'
    Expected type: Pi Account String
      Actual type: Pi Account (Maybe String)
    In the second argument of `(!)', namely Account.productCd' 
    In the second argument of `(|*|)', namely a ! Account.productCd'
    In the first argument of `(|*|)', namely
      Account1 |$| a ! Account.accountId' |*| a ! Account.productCd'

src/examples.hs:372:29:
    Couldn't match type `Maybe Int64' with `Int64'
    Expected type: Pi Account Int64
      Actual type: Pi Account (Maybe Int64)
    In the second argument of `(!)', namely Account.custId'
    In the second argument of `(|*|)', namely a ! Account.custId'
    In the first argument of `(|*|)', namely
      Account1 |$| a ! Account.accountId' |*| a ! Account.productCd'
      |*| a ! Account.custId'
ghc --make: /usr/hs/ghc/7.6.3/bin/ghc failure (return code=1)  
cabal: Error: some packages failed to install:
relational-record-examples-0.2.0.0 failed during the building phase. The
exception was:
ExitFailure 1
```

Running `cabal install --ghc-options=-ddump-splices` on Account.hs, I get the following output:

```
src/Account.hs:1:1: Splicing declarations
    defineTable "account"
  ======>
    src/Account.hs:7:3-23
    data Account
      = Account {accountId :: !(Maybe GHC.Int.Int64),
                 productCd :: !(Maybe String),
                 custId :: !(Maybe GHC.Int.Int64),
                 openDate :: !(Maybe time-1.4.0.1:Data.Time.Calendar.Days.Day),
                 closeDate :: !(Maybe time-1.4.0.1:Data.Time.Calendar.Days.Day),
                 lastActivityDate :: !(Maybe time-1.4.0.1:Data.Time.Calendar.Days.Day),
                 status :: !(Maybe String),
                 openBranchId :: !(Maybe GHC.Int.Int64),
                 openEmpId :: !(Maybe GHC.Int.Int64),
                 availBalance :: !(Maybe Double),
                 pendingBalance :: !(Maybe Double)}
      deriving (Show)
    columnOffsetsAccount :: GHC.Arr.Array Int Int
    columnOffsetsAccount
      = ((GHC.Arr.listArray (0 :: Int, 11))
         $ (scanl
              (+)
              (0 :: Int)
              [Database.Record.Persistable.runPersistableRecordWidth
                 (Database.Record.Persistable.persistableWidth ::
                    Database.Record.Persistable.PersistableRecordWidth (Maybe GHC.Int.Int64)),
               Database.Record.Persistable.runPersistableRecordWidth
                 (Database.Record.Persistable.persistableWidth ::
                    Database.Record.Persistable.PersistableRecordWidth (Maybe String)),
               Database.Record.Persistable.runPersistableRecordWidth
                 (Database.Record.Persistable.persistableWidth ::
                    Database.Record.Persistable.PersistableRecordWidth (Maybe GHC.Int.Int64)),
               Database.Record.Persistable.runPersistableRecordWidth
                 (Database.Record.Persistable.persistableWidth ::
                    Database.Record.Persistable.PersistableRecordWidth (Maybe time-1.4.0.1:Data.Time.Calendar.Days.Day)),
               Database.Record.Persistable.runPersistableRecordWidth
                 (Database.Record.Persistable.persistableWidth ::
                    Database.Record.Persistable.PersistableRecordWidth (Maybe time-1.4.0.1:Data.Time.Calendar.Days.Day)),
               Database.Record.Persistable.runPersistableRecordWidth
                 (Database.Record.Persistable.persistableWidth ::
                    Database.Record.Persistable.PersistableRecordWidth (Maybe time-1.4.0.1:Data.Time.Calendar.Days.Day)),
               Database.Record.Persistable.runPersistableRecordWidth
                 (Database.Record.Persistable.persistableWidth ::
                    Database.Record.Persistable.PersistableRecordWidth (Maybe String)),
               Database.Record.Persistable.runPersistableRecordWidth
                 (Database.Record.Persistable.persistableWidth ::
                    Database.Record.Persistable.PersistableRecordWidth (Maybe GHC.Int.Int64)),
               Database.Record.Persistable.runPersistableRecordWidth
                 (Database.Record.Persistable.persistableWidth ::
                    Database.Record.Persistable.PersistableRecordWidth (Maybe GHC.Int.Int64)),
               Database.Record.Persistable.runPersistableRecordWidth
                 (Database.Record.Persistable.persistableWidth ::
                    Database.Record.Persistable.PersistableRecordWidth (Maybe Double)),
               Database.Record.Persistable.runPersistableRecordWidth
                 (Database.Record.Persistable.persistableWidth ::
                    Database.Record.Persistable.PersistableRecordWidth (Maybe Double))]))
    instance Database.Record.Persistable.PersistableWidth Account where
      Database.Record.Persistable.persistableWidth
        = (Database.Record.Persistable.unsafePersistableRecordWidth
           $ (columnOffsetsAccount GHC.Arr.! 11))
    instance Database.Relational.Query.Pure.ProductConstructor (Maybe GHC.Int.Int64
                                                                -> Maybe String
                                                                   -> Maybe GHC.Int.Int64
                                                                      -> Maybe time-1.4.0.1:Data.Time.Calendar.Days.Day
                                                                         -> Maybe time-1.4.0.1:Data.Time.Calendar.Days.Day
                                                                            -> Maybe time-1.4.0.1:Data.Time.Calendar.Days.Day
                                                                               -> Maybe String
                                                                                  -> Maybe GHC.Int.Int64
                                                                                     -> Maybe GHC.Int.Int64
                                                                                        -> Maybe Double
                                                                                           -> Maybe Double
                                                                                              -> Account) where
      Database.Relational.Query.Pure.productConstructor = Account
    instance Database.Relational.Query.Table.TableDerivable Account where
      Database.Relational.Query.Table.derivedTable
        = Database.Relational.Query.Table.table
            "MAIN.account"
            ["account_id", "product_cd", "cust_id", "open_date", "close_date",
             "last_activity_date", "status", "open_branch_id", "open_emp_id",
             "avail_balance", "pending_balance"]
    tableOfAccount :: Database.Relational.Query.Table.Table Account
    tableOfAccount = Database.Relational.Query.Table.derivedTable
    account :: Database.Relational.Query.Relation.Relation () Account
    account = Database.Relational.Query.Relation.derivedRelation
    insertAccount :: Database.Relational.Query.Type.Insert Account
    insertAccount
      = Database.Relational.Query.Type.derivedInsert
          Database.Relational.Query.Pi.id'
    insertQueryAccount ::
      forall p_a3kd.
      Database.Relational.Query.Relation.Relation p_a3kd Account
      -> Database.Relational.Query.Type.InsertQuery p_a3kd
    insertQueryAccount
      = Database.Relational.Query.Type.derivedInsertQuery
          Database.Relational.Query.Pi.id'
    accountId' ::
      Database.Relational.Query.Pi.Unsafe.Pi Account (Maybe GHC.Int.Int64)
    accountId'
      = Database.Relational.Query.Pi.Unsafe.definePi
          (columnOffsetsAccount array-0.4.0.1:Data.Array.Base.! 0)
    productCd' ::
      Database.Relational.Query.Pi.Unsafe.Pi Account (Maybe String)
    productCd'
      = Database.Relational.Query.Pi.Unsafe.definePi
          (columnOffsetsAccount array-0.4.0.1:Data.Array.Base.! 1)
    custId' ::
      Database.Relational.Query.Pi.Unsafe.Pi Account (Maybe GHC.Int.Int64)
    custId'
      = Database.Relational.Query.Pi.Unsafe.definePi
          (columnOffsetsAccount array-0.4.0.1:Data.Array.Base.! 2)
    openDate' ::
      Database.Relational.Query.Pi.Unsafe.Pi Account (Maybe time-1.4.0.1:Data.Time.Calendar.Days.Day)
    openDate'
      = Database.Relational.Query.Pi.Unsafe.definePi
          (columnOffsetsAccount array-0.4.0.1:Data.Array.Base.! 3)
    closeDate' ::
      Database.Relational.Query.Pi.Unsafe.Pi Account (Maybe time-1.4.0.1:Data.Time.Calendar.Days.Day)
    closeDate'
      = Database.Relational.Query.Pi.Unsafe.definePi
          (columnOffsetsAccount array-0.4.0.1:Data.Array.Base.! 4)
    lastActivityDate' ::
      Database.Relational.Query.Pi.Unsafe.Pi Account (Maybe time-1.4.0.1:Data.Time.Calendar.Days.Day)
    lastActivityDate'
      = Database.Relational.Query.Pi.Unsafe.definePi
          (columnOffsetsAccount array-0.4.0.1:Data.Array.Base.! 5)
    status' ::
      Database.Relational.Query.Pi.Unsafe.Pi Account (Maybe String)
    status'
      = Database.Relational.Query.Pi.Unsafe.definePi
          (columnOffsetsAccount array-0.4.0.1:Data.Array.Base.! 6)
    openBranchId' ::
      Database.Relational.Query.Pi.Unsafe.Pi Account (Maybe GHC.Int.Int64)
    openBranchId'
      = Database.Relational.Query.Pi.Unsafe.definePi
          (columnOffsetsAccount array-0.4.0.1:Data.Array.Base.! 7)
    openEmpId' ::
      Database.Relational.Query.Pi.Unsafe.Pi Account (Maybe GHC.Int.Int64)
    openEmpId'
      = Database.Relational.Query.Pi.Unsafe.definePi
          (columnOffsetsAccount array-0.4.0.1:Data.Array.Base.! 8)
    availBalance' ::
      Database.Relational.Query.Pi.Unsafe.Pi Account (Maybe Double)
    availBalance'
      = Database.Relational.Query.Pi.Unsafe.definePi
          (columnOffsetsAccount array-0.4.0.1:Data.Array.Base.! 9)
    pendingBalance' ::
      Database.Relational.Query.Pi.Unsafe.Pi Account (Maybe Double)
    pendingBalance'
      = Database.Relational.Query.Pi.Unsafe.definePi
          (columnOffsetsAccount array-0.4.0.1:Data.Array.Base.! 10)
    instance Database.Record.KeyConstraint.HasColumnConstraint Database.Record.KeyConstraint.Primary Account where
      Database.Record.KeyConstraint.columnConstraint
        = Database.Record.KeyConstraint.unsafeSpecifyColumnConstraint 0
    instance Database.Record.KeyConstraint.HasKeyConstraint Database.Record.KeyConstraint.Primary Account where
      Database.Record.KeyConstraint.keyConstraint
        = Database.Record.KeyConstraint.derivedCompositePrimary
    instance Database.Relational.Query.Constraint.HasConstraintKey Database.Record.KeyConstraint.Primary Account (Maybe GHC.Int.Int64) where
      Database.Relational.Query.Constraint.constraintKey
        = Database.Relational.Query.Constraint.unsafeDefineConstraintKey
            [0]
    selectAccount ::
      Database.Relational.Query.Type.Query (Maybe GHC.Int.Int64) Account
    selectAccount
      = Database.Relational.Query.Type.relationalQuery
          (Database.Relational.Query.Derives.primary account)
    updateAccount ::
      Database.Relational.Query.Type.KeyUpdate (Maybe GHC.Int.Int64) Account
    updateAccount
      = Database.Relational.Query.Derives.primaryUpdate tableOfAccount
    fromSqlOfAccount ::
      Database.Record.FromSql.RecordFromSql Database.HDBC.SqlValue.SqlValue Account
    fromSqlOfAccount
      = ((((((((((((Control.Applicative.pure Account)
                   Control.Applicative.<*> Database.Record.FromSql.recordFromSql)
                  Control.Applicative.<*> Database.Record.FromSql.recordFromSql)
                 Control.Applicative.<*> Database.Record.FromSql.recordFromSql)
                Control.Applicative.<*> Database.Record.FromSql.recordFromSql)
               Control.Applicative.<*> Database.Record.FromSql.recordFromSql)
              Control.Applicative.<*> Database.Record.FromSql.recordFromSql)
             Control.Applicative.<*> Database.Record.FromSql.recordFromSql)
            Control.Applicative.<*> Database.Record.FromSql.recordFromSql)
           Control.Applicative.<*> Database.Record.FromSql.recordFromSql)
          Control.Applicative.<*> Database.Record.FromSql.recordFromSql)
         Control.Applicative.<*> Database.Record.FromSql.recordFromSql)
    toSqlOfAccount ::
      Database.Record.ToSql.RecordToSql Database.HDBC.SqlValue.SqlValue Account
    toSqlOfAccount
      = Database.Record.ToSql.wrapToSql
          (\ (Account f1_a3ke
                      f2_a3kf
                      f3_a3kg
                      f4_a3kh
                      f5_a3ki
                      f6_a3kj
                      f7_a3kk
                      f8_a3kl
                      f9_a3km
                      f10_a3kn
                      f11_a3ko)
             -> ((Database.Record.ToSql.putRecord f1_a3ke)
                 >>
                   ((Database.Record.ToSql.putRecord f2_a3kf)
                    >>
                      ((Database.Record.ToSql.putRecord f3_a3kg)
                       >>
                         ((Database.Record.ToSql.putRecord f4_a3kh)
                          >>
                            ((Database.Record.ToSql.putRecord f5_a3ki)
                             >>
                               ((Database.Record.ToSql.putRecord f6_a3kj)
                                >>
                                  ((Database.Record.ToSql.putRecord f7_a3kk)
                                   >>
                                     ((Database.Record.ToSql.putRecord f8_a3kl)
                                      >>
                                        ((Database.Record.ToSql.putRecord f9_a3km)
                                         >>
                                           ((Database.Record.ToSql.putRecord f10_a3kn)
                                            >>
                                              ((Database.Record.ToSql.putRecord f11_a3ko)
                                               >>
                                                 (Database.Record.ToSql.putEmpty
                                                    ghc-prim:GHC.Tuple.())))))))))))))
    instance Database.Record.FromSql.FromSql Database.HDBC.SqlValue.SqlValue Account where
      Database.Record.FromSql.recordFromSql = fromSqlOfAccount
    instance Database.Record.ToSql.ToSql Database.HDBC.SqlValue.SqlValue Account where
      Database.Record.ToSql.recordToSql = toSqlOfAccount
```

It looks like the Account record's types are being created as `Maybe`, but `example.hs` expects them without `Maybe`.  

I am running on an old system (CentOS 5), so maybe something could be going wrong with the interaction between the system sqlite library and HDBC sqlite driver?

Here is my system version:

```
$ cat /etc/redhat-release
CentOS release 5.11 (Final)
$ uname -a
Linux machine 2.6.18-400.1.1.el5xen #1 SMP Thu Dec 18 01:40:36 EST 2014 x86_64 x86_64 x86_64 GNU/Linux
```

Here are the versions of sqlite3 I have installed on my system:

```
$ rpm -qa | grep -i sqlite
sqlite-3.3.6-7
sqlite-devel-3.3.6-7
```

Here are the versions of packages I have installed in my sandbox when trying to build the `example.hs` file from relational-record-examples:

```
$ cabal sandbox hc-pkg list
/usr/hs/ghc/7.6.3/lib/ghc-7.6.3/package.conf.d
   Cabal-1.16.0
   array-0.4.0.1
   base-4.6.0.1
   bin-package-db-0.0.0.0
   binary-0.5.1.1
   bytestring-0.10.0.2
   containers-0.5.0.0
   deepseq-1.3.0.1
   directory-1.2.0.1
   filepath-1.3.0.1
   ghc-7.6.3
   ghc-prim-0.3.0.0
   haskell2010-1.1.1.0
   haskell98-2.0.0.2
   hoopl-3.9.0.0
   hpc-0.6.0.0
   integer-gmp-0.5.0.0
   old-locale-1.0.0.5
   old-time-1.1.0.1
   pretty-1.1.1.0
   process-1.1.0.2
   rts-1.0
   template-haskell-2.8.0.0
   time-1.4.0.1
   unix-2.6.0.1
/home/me/git/relational-record-examples-0.2.0.0/.cabal-sandbox/x86_64-linux-ghc-7.6.3-packages.conf.d
   HDBC-2.4.0.0
   HDBC-session-0.1.0.0
   HDBC-sqlite3-2.3.3.0
   convertible-1.1.0.0
   dlist-0.7.1
   mtl-2.2.1
   names-th-0.1.0.0
   persistable-record-0.1.0.0
   relational-query-0.4.0.1
   relational-query-HDBC-0.1.0.0
   relational-schemas-0.1.0.1
   sql-words-0.1.3.0
   text-1.2.0.4
   time-locale-compat-0.1.0.1
   transformers-0.4.3.0
   utf8-string-1
```

Here is a much shorter version of `example.hs` where this problem is reproducable:

```
{-# LANGUAGE FlexibleContexts #-}

import Database.Record (FromSql, ToSql)
import Database.Relational.Query ((!), in', query, Relation, relation, relationalQuery, wheres, values)

import Database.HDBC (IConnection, SqlValue, rollback)
import Database.HDBC.Record (runQuery)
import Database.HDBC.Session (withConnectionIO, handleSqlError')

import qualified Account
import Account (Account, account, tableOfAccount)

import DataSource (connect)

account_4_3_3a :: Relation () Account
account_4_3_3a = relation $ do
  a  <- query account
  wheres $ a ! Account.productCd' `in'` values ["CHK", "SAV", "CD", "MM"]
  return a

run :: (Show a, IConnection conn, FromSql SqlValue a, ToSql SqlValue p)
    => conn -> p -> Relation p a -> IO ()
run conn param rel = do
  putStrLn $ "SQL: " ++ show rel
  records <- runQuery conn (relationalQuery rel) param
  mapM_ print records
  putStrLn ""
  rollback conn

main :: IO ()
main = handleSqlError' $ withConnectionIO connect $ \conn ->
  run conn () account_4_3_3a
```

Here is the error that occurs in this shorter version:

```
$ cabal install
Resolving dependencies...
Configuring relational-record-examples-0.2.0.0...
Building relational-record-examples-0.2.0.0...
Preprocessing executable 'examples' for relational-record-examples-0.2.0.0...
[3 of 3] Compiling Main             ( src/examples.hs, dist/dist-sandbox-3657a918/build/examples/examples-tmp/Main.o )

src/examples.hs:18:49:
    Couldn't match expected type `Maybe String'
                with actual type `[Char]'
    In the expression: "CHK"
    In the first argument of `values', namely
      `["CHK", "SAV", "CD", "MM"]'
    In the second argument of in', namely
      `values ["CHK", "SAV", "CD", "MM"]'

src/examples.hs:18:56:
    Couldn't match expected type `Maybe String'
                with actual type `[Char]'
    In the expression: "SAV"
    In the first argument of `values', namely
      `["CHK", "SAV", "CD", "MM"]'
    In the second argument of in', namely
      `values ["CHK", "SAV", "CD", "MM"]'

src/examples.hs:18:63:
    Couldn't match expected type `Maybe String'
                with actual type `[Char]'
    In the expression: "CD"
    In the first argument of `values', namely
      `["CHK", "SAV", "CD", "MM"]'
    In the second argument of in', namely
      `values ["CHK", "SAV", "CD", "MM"]'

src/examples.hs:18:69:
    Couldn't match expected type `Maybe String'
                with actual type `[Char]'
    In the expression: "MM"
    In the first argument of `values', namely
      `["CHK", "SAV", "CD", "MM"]'
    In the second argument of in', namely
      `values ["CHK", "SAV", "CD", "MM"]'
ghc --make: /usr/hs/ghc/7.6.3/bin/ghc failure (return code=1)
Failed to install relational-record-examples-0.2.0.0
cabal: Error: some packages failed to install:
relational-record-examples-0.2.0.0 failed during the building phase. The
exception was:
ExitFailure 1
```

Here is the same version that works.  You can see where I added `Just` to get rid of the problem:

```
{-# LANGUAGE FlexibleContexts #-}

import Database.Record (FromSql, ToSql)
import Database.Relational.Query ((!), in', query, Relation, relation, relationalQuery, wheres, values)

import Database.HDBC (IConnection, SqlValue, rollback)
import Database.HDBC.Record (runQuery)
import Database.HDBC.Session (withConnectionIO, handleSqlError')

import qualified Account
import Account (Account, account, tableOfAccount)

import DataSource (connect)

account_4_3_3a :: Relation () Account
account_4_3_3a = relation $ do
  a  <- query account
  wheres $ a ! Account.productCd' `in'` values [Just "CHK", Just "SAV", Just "CD", Just "MM"]
  return a

run :: (Show a, IConnection conn, FromSql SqlValue a, ToSql SqlValue p)
    => conn -> p -> Relation p a -> IO ()
run conn param rel = do
  putStrLn $ "SQL: " ++ show rel
  records <- runQuery conn (relationalQuery rel) param
  mapM_ print records
  putStrLn ""
  rollback conn

main :: IO ()
main = handleSqlError' $ withConnectionIO connect $ \conn ->
  run conn () account_4_3_3a
```

**Comment by yuga** [comment]
I consider this issue is caused by the incompatibility of pragma table_info between old sqlite3 and current sqlite3.
I will fix it.

**Comment by kazu-yamamoto** [comment]
@cdepillabout Does the patch above fix this problem?

If so, let's release a new version. > @khibino

**Comment by cdepillabout** [comment]
This does fix the above problem!  Thanks!

**Comment by yuga** [comment]
I have released relational-schemas-0.1.0.2.
@cdepillabout Thank you for your report.

**Comment by yuga** [comment]
memo:

- sqlite-3.8.8.3 returns 1 or 0 as notnull.
```c
  /*
  **   PRAGMA table_info(<table>)
  **
  ** Return a single row for each column of the named table. The columns of
  ** the returned data set are:
  **
  ** cid:        Column id (numbered from left to right, starting at 0)
  ** name:       Column name
  ** type:       Column declaration type.
  ** notnull:    True if 'NOT NULL' is part of column declaration
  ** dflt_value: The default value for the column, if any.
  */
  case PragTyp_TABLE_INFO: if( zRight ){
    Table *pTab;
    pTab = sqlite3FindTable(db, zRight, zDb);
    if( pTab ){
      int i, k;
      int nHidden = 0;
      Column *pCol;
      Index *pPk = sqlite3PrimaryKeyIndex(pTab);
      sqlite3VdbeSetNumCols(v, 6);
      pParse->nMem = 6;
      sqlite3CodeVerifySchema(pParse, iDb);
      sqlite3VdbeSetColName(v, 0, COLNAME_NAME, "cid", SQLITE_STATIC);
      sqlite3VdbeSetColName(v, 1, COLNAME_NAME, "name", SQLITE_STATIC);
      sqlite3VdbeSetColName(v, 2, COLNAME_NAME, "type", SQLITE_STATIC);
      sqlite3VdbeSetColName(v, 3, COLNAME_NAME, "notnull", SQLITE_STATIC);
      sqlite3VdbeSetColName(v, 4, COLNAME_NAME, "dflt_value", SQLITE_STATIC);
      sqlite3VdbeSetColName(v, 5, COLNAME_NAME, "pk", SQLITE_STATIC);
      sqlite3ViewGetColumnNames(pParse, pTab);
      for(i=0, pCol=pTab->aCol; i<pTab->nCol; i++, pCol++){
        if( IsHiddenColumn(pCol) ){
          nHidden++;
          continue;
        }
        sqlite3VdbeAddOp2(v, OP_Integer, i-nHidden, 1);
        sqlite3VdbeAddOp4(v, OP_String8, 0, 2, 0, pCol->zName, 0);
        sqlite3VdbeAddOp4(v, OP_String8, 0, 3, 0,
           pCol->zType ? pCol->zType : "", 0);
        sqlite3VdbeAddOp2(v, OP_Integer, (pCol->notNull ? 1 : 0), 4);
        if( pCol->zDflt ){
          sqlite3VdbeAddOp4(v, OP_String8, 0, 5, 0, (char*)pCol->zDflt, 0);
        }else{
          sqlite3VdbeAddOp2(v, OP_Null, 0, 5);
        }
        if( (pCol->colFlags & COLFLAG_PRIMKEY)==0 ){
          k = 0;
        }else if( pPk==0 ){
          k = 1;
        }else{
          for(k=1; ALWAYS(k<=pTab->nCol) && pPk->aiColumn[k-1]!=i; k++){}
        }
        sqlite3VdbeAddOp2(v, OP_Integer, k, 6);
        sqlite3VdbeAddOp2(v, OP_ResultRow, 1, 6);
      }
    }
  }
  break;
```

- sqlite-3.3.8 directly returns u8 value as notnull.
```c
  /*
  **   PRAGMA table_info(<table>)
  **
  ** Return a single row for each column of the named table. The columns of
  ** the returned data set are:
  **
  ** cid:        Column id (numbered from left to right, starting at 0)
  ** name:       Column name
  ** type:       Column declaration type.
  ** notnull:    True if 'NOT NULL' is part of column declaration
  ** dflt_value: The default value for the column, if any.
  */
  if( sqlite3StrICmp(zLeft, "table_info")==0 && zRight ){
    Table *pTab;
    if( sqlite3ReadSchema(pParse) ) goto pragma_out;
    pTab = sqlite3FindTable(db, zRight, zDb);
    if( pTab ){
      int i;
      Column *pCol;
      sqlite3VdbeSetNumCols(v, 6);
      sqlite3VdbeSetColName(v, 0, COLNAME_NAME, "cid", P3_STATIC);
      sqlite3VdbeSetColName(v, 1, COLNAME_NAME, "name", P3_STATIC);
      sqlite3VdbeSetColName(v, 2, COLNAME_NAME, "type", P3_STATIC);
      sqlite3VdbeSetColName(v, 3, COLNAME_NAME, "notnull", P3_STATIC);
      sqlite3VdbeSetColName(v, 4, COLNAME_NAME, "dflt_value", P3_STATIC);
      sqlite3VdbeSetColName(v, 5, COLNAME_NAME, "pk", P3_STATIC);
      sqlite3ViewGetColumnNames(pParse, pTab);
      for(i=0, pCol=pTab->aCol; i<pTab->nCol; i++, pCol++){
        const Token *pDflt;
        static const Token noDflt =  { (unsigned char*)"", 0, 0 };
        sqlite3VdbeAddOp(v, OP_Integer, i, 0);
        sqlite3VdbeOp3(v, OP_String8, 0, 0, pCol->zName, 0);
        sqlite3VdbeOp3(v, OP_String8, 0, 0,
           pCol->zType ? pCol->zType : "", 0);
        sqlite3VdbeAddOp(v, OP_Integer, pCol->notNull, 0);
        pDflt = pCol->pDflt ? &pCol->pDflt->span : &noDflt;
        if( pDflt->z ){
          sqlite3VdbeOp3(v, OP_String8, 0, 0, (char*)pDflt->z, pDflt->n);
        }else{
          sqlite3VdbeAddOp(v, OP_Null, 0, 0);
        }
        sqlite3VdbeAddOp(v, OP_Integer, pCol->isPrimKey, 0);
        sqlite3VdbeAddOp(v, OP_Callback, 6, 0);
      }
    }
  }else
```
